docs/design/design.md

```diff
@@ -135,56 +192,82 @@ Other flags that are set:
  - `--requestheader-client-ca-file` to `front-proxy-ca.crt`
  - `--admission-control` to `NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota`
    - ...or whatever the recommended set of admission controllers is at a given version
- - `--storage-backend` to `etcd3`. Support for `etcd2` in kubeadm is dropped.
  - `--kubelet-preferred-address-types` to `InternalIP,ExternalIP,Hostname`
    - This makes `kubectl logs` and other apiserver -> kubelet communication work in environments where the hostnames of the nodes aren't resolvable
- - `--requestheader-username-headers=X-Remote-User`, `--requestheader-group-headers=X-Remote-Group`, `--requestheader-extra-headers-prefix=X-Remote-Extra-`, --requestheader-allowed-names=front-proxy-client` so the front proxy (API Aggregation) communication is secure.
+ - `--requestheader-username-headers=X-Remote-User`, `--requestheader-group-headers=X-Remote-Group`, `--requestheader-extra-headers-prefix=X-Remote-Extra-`, `--requestheader-allowed-names=front-proxy-client` so the front proxy (API Aggregation) communication is secure.
+ - `--allow-privileged` to `true` 
+ - ``--kubelet-preferred-address-types` to `InternalIP,ExternalIP,Hostname`
 
 
 #### Controller Manager
 
 The controller-manager needs to know this in particular:
  - The Pod Network CIDR if any; also enables the Subnet Manager feature (required for some CNI network plugins)
+ - Eventually the cloud provider specified by the user
 
 Other flags that are set unconditionally:
  - The `BootstrapSigner` and `TokenCleaner` controllers are enabled
  - `--root-ca-file` to `ca.crt`
- - `--cluster-signing-cert-file` to `ca.crt`
- - `--cluster-signing-key-file` to `ca.key`
+ - `--cluster-signing-cert-file` to `ca.crt`, expect  if `kubeadm` is running in "ExternalCA"
+ - `--cluster-signing-key-file` to `ca.key`, expect  if `kubeadm` is running in "ExternalCA"
  - `--service-account-private-key-file` to `sa.key`
  - `--use-service-account-credentials` to `true`
 
+Please note that:
+
+1. if `kubeadm` is running in "ExternalCA" mode, `--cluster-signing-cert-file` and  `--cluster-signing-key-file` will be set to  `""` .
+
 #### Scheduler
 
 kubeadm doesn't set any special scheduler flags.
 
-Common properties for the control plane components:
- - Leader election is enabled for both the controller-manager and the scheduler
- - `HostNetwork: true` is present on all static pods since there is no network configured yet
+### Generate Static Pod Manifest for local etcd
```

Comment:
This happens right before the controlplane phase, right?

Reply:
in general, can you add the `kubeadm phase` command for each of these phases.
e.g. in this case `kubeadm phase etcd local` in this case...

Reply:
Currently controlplane manifest are created before the etcd manifest.